modelon/impact/client/entities/case.py

```diff
@@ -127,6 +129,89 @@ def simulation_log_level(self, simulation_log_level: str):
         self._analysis['simulation_log_level'] = simulation_log_level
 
 
+class CustomArtifact:
+    """CustomArtifact class"""
+
+    def __init__(
+        self,
+        workspace_id: str,
+        experiment_id: str,
+        case_id: str,
+        artifact_id: str,
+        download_as: str,
+        exp_sal,
+    ):
+        self._workspace_id = workspace_id
+        self._exp_id = experiment_id
+        self._case_id = case_id
+        self._artifact_id = artifact_id
+        self._download_as = download_as
+        self._exp_sal = exp_sal
+
+    @property
+    def id(self):
+        """Id of the custom artifact."""
+        return self._artifact_id
+
+    @property
+    def download_as(self):
+        """File name for the downloaded artifact."""
+        return self._download_as
+
+    def download(self, path: Optional[str] = None):
+        """Downloads a custom artifact.
+        Returns the local path to the downloaded artifact.
+
+        Parameters:
+
+            path --
+                The local path to the directory to store the downloaded custom artifact.
+                Default: None. If no path is given, custom artifact will be downloaded
+                in a temporary directory.
+
+        Returns:
+
+            path --
+                Local path to the downloaded custom artifact.
+
+        Example::
+
+            artifact_path = artifact.download()
+            artifact_path = artifact.download('/home/Downloads')
+        """
+        artifact, _ = self._exp_sal.case_artifact_get(
+            self._workspace_id, self._exp_id, self._case_id, self.id
+        )
+        if path is None:
+            path = os.path.join(tempfile.gettempdir(), "impact-downloads")
+        os.makedirs(path, exist_ok=True)
+        artifact_path = os.path.join(path, self.download_as)
+        with open(artifact_path, "wb") as f:
+            f.write(artifact)
+        return artifact_path
+
+    def to_stream(self) -> bytes:
+        """
```

Comment:
`get_data` per previous comment. Streams are really about asyncio and this is not the case.